File: q1.py
import sys
def sumOfDigit(n) :  
    sum = 0;  
    while (n > 0) : 
        sum += n % 10;  
        n = n // 10;  
    return sum;  

def sortArr(arr, n) : 
    vp = [];  
    for i in range(n) : 
        vp.append([sumOfDigit(arr[i]), arr[i]]);   
    vp.sort() 
    for i in range(len(vp)) : 
        print(vp[i][1], end = " "); 

if __name__ == "__main__" :   
    n = int(input())
    arr = []
    for i in range(n):
        x = [int(i) for i in input().split()]
        arr.append(x)  
    sortArr(arr, n);  

File: q1.cpp

#include <bits/stdc++.h> 
using namespace std; 
int sumOfDigit(int n) { 
    int sum = 0; 
    while (n > 0) { 
        sum += n % 10; 
        n = n / 10; 
    } 
    return sum; 
}  
void sortArr(int arr[], int n) {  
    vector<pair<int, int> > vp; 
    for (int i = 0; i < n; i++) { 
        vp.push_back(make_pair(sumOfDigit(arr[i]), arr[i])); 
    }
    sort(vp.begin(), vp.end());  
    for (int i = 0; i < vp.size(); i++) 
        cout << vp[i].second << " "; 
} 
int main() { 
    int n; 
    cin >> n;
    int *arr = new int[n];
    for(int i=0;i<n;i++){
        cin >> arr[i];
    }
    sortArr(arr, n); 
    return 0; 
} 